assets/js/lazyload-scripts.js

```diff
@@ -54,7 +54,12 @@ class RocketLazyLoadScripts {
 	 * load the script.
 	 */
 	_loadScriptSrc() {
-		const scripts = document.querySelectorAll( `script[${ this.attrName }]` );
+		const scripts = Array.prototype.slice.call(
+			document.querySelectorAll(
+				`script[${ this.attrName }]`
+			)
```

Comment:
This is safe to do here. Why? If there are no scripts with the `data-rocketlazyloadscript` attribute, `querySelectorAll` returns an empty `NodeList`. 

[Per the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#Return_value):

>Return Value
>A non-live NodeList containing one Element object for each element that matches at least one of the specified selectors or an empty NodeList in case of no matches.

However, it's not as efficient as it could be. Why? If there are no scripts, there's no need to do the slice or the `forEach`. Therefore, a tiny little micro-optimization would be:

```
const scripts = document.querySelectorAll( `script[${ this.attrName }]` );

if ( 0 === scripts ) {
     return;
}

Array.prototype.slice.call( scripts ).forEach( elem => {          
```

Reply:
Good idea, I also called `this.reset();` before bailing out so removing the event listener as it's not needed anymore.